doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+address of the global variable (2).
+
+Besides that, jextract also generates a getter and a setter method to get and set the value
+of the global variable (3). Once again, the parameter and return type of the getter and
+setter depend on the carrier type of the layout of the global variable, which is itself
+derived from the header files.
+
+### Constants (Macros & Enums)
+
+Both macros and enums are translated similarly. If we have a header file containing these
+declarations:
+
+```c
+// mylib.h
+
+#define MY_MACRO 42
+enum MY_ENUM { A, B, C };
+```
+
+Jextract will generate a set of simple getter methods to access the constant values of the
+macro and the enum constants:
+
+```java
+// mylib_h.java
+
+public static int MY_MACRO() { ... }
+
+public static int A() { ... }
+public static int B() { ... }
+public static int C() { ... }
+```
+
+Note that the enum constants are exposed as top-level methods, rather than being nested
+inside a class called `MY_ENUM`, or through the use of a Java `enum`. This translation
+strategy mimics C's behavior of enum constants being accessible as a top-level declaration
+as well.
+
+Not all types of macros are supported though. Only macros that have a primitive numerical
+value, a string, or a pointer type are supported. Most notably, function-like macros are
+not supported by jextract.
+
+### Structs & Unions
+
+Things get a little more complicated for structs and unions. For a struct delcaration like
+this:
+
+```c
+// mylib.h
+
+struct Point {
+    int x;
+    int y;
+};
+```
+
+Jextract generates a separate class which roughly looks like the following:
+
+```java
+// Point.java
+
+public class Point {
+    public static final GroupLayout layout() { ... } // 3
+
+    public static final OfInt x$layout() { ... } // 2
+    public static final long x$offset() { ... } // 2
+
+    public static int x(MemorySegment struct) { ... } // 1
+    public static void x(MemorySegment struct, int fieldValue) { ... } // 1
+
+    public static final OfInt y$layout() { ... } // 2
+    public static final long y$offset() { ... } // 2
+
+    public static int y(MemorySegment struct) { ... } // 1
+    public static void y(MemorySegment struct, int fieldValue) { ... } // 1
+
+    public static MemorySegment asSlice(MemorySegment array, long index) { ... } // 5
+
+    public static long sizeof() { ... } // 3
+
+    public static MemorySegment allocate(SegmentAllocator allocator) { ... } // 4
+    public static MemorySegment allocateArray(long elementCount,
+            SegmentAllocator allocator) { ... } // 4
+
+    public static MemorySegment reinterpret(MemorySegment addr, Arena arena,
+            Consumer<MemorySegment> cleanup) { ... } // 6
+    public static MemorySegment reinterpret(MemorySegment addr, long elementCount,
+            Arena arena, Consumer<MemorySegment> cleanup) { ... } // 6
+}
+```
+
+There's a getter and setter for each field of the struct (1), which takes a pointer to a struct
```

Comment:
I see you didn't use bullet lists much, but I wonder if one would be helpful here instead of a single long para?

Reply:
I think it would be appropriate here